src/main/java/cc/altius/FASP/security/CustomUserDetailsService.java

```diff
@@ -35,6 +39,12 @@ public CustomUserDetails loadUserByUsername(String emailId) throws UsernameNotFo
             if (user == null || !user.isPresent()) {
                 throw new UsernameNotFoundException("User not found");
             }
+            if (testMode) {
+                CustomUserDetails testUser = this.userDao.getCustomUserByUserId(user.getUserId());
+                if (testUser != null) {
+                    return testUser;
+                }
+            }
```

Comment:
I'm a little concerned by the leak of test specific code into production code. I know we discussed this, but I am wondering if there isn't a way around this. Could we have a different implementation of UserDetailsService for the test environment?